aws/resource_aws_efs_access_point.go

```diff
@@ -0,0 +1,404 @@
+							Type:     schema.TypeInt,
+							Required: true,
+							ForceNew: true,
+						},
+						"uid": {
+							Type:     schema.TypeInt,
+							Required: true,
+							ForceNew: true,
+						},
+						"secondary_gids": {
+							Type:     schema.TypeSet,
+							Elem:     &schema.Schema{Type: schema.TypeInt},
+							Set:      schema.HashInt,
+							Optional: true,
+							ForceNew: true,
+						},
+					},
+				},
+			},
+			"root_directory": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				ForceNew: true,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"path": {
+							Type:     schema.TypeString,
+							Optional: true,
+							ForceNew: true,
+							Computed: true,
+						},
+						"creation_info": {
+							Type:     schema.TypeList,
+							Optional: true,
+							ForceNew: true,
+							MaxItems: 1,
+							Computed: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"owner_gid": {
+										Type:     schema.TypeInt,
+										Required: true,
+										ForceNew: true,
+									},
+									"owner_uid": {
+										Type:     schema.TypeInt,
+										Required: true,
+										ForceNew: true,
+									},
+									"permissions": {
+										Type:     schema.TypeString,
+										Required: true,
+										ForceNew: true,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			"tags": tagsSchema(),
+		},
+	}
+}
+
+func resourceAwsEfsAccessPointCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).efsconn
+
+	fsId := d.Get("file_system_id").(string)
+
+	input := efs.CreateAccessPointInput{
+		FileSystemId: aws.String(fsId),
+		Tags:         keyvaluetags.New(d.Get("tags").(map[string]interface{})).IgnoreAws().EfsTags(),
+	}
+
+	if v, ok := d.GetOk("posix_user"); ok {
+		input.PosixUser = expandEfsAccessPointPosixUser(v.([]interface{}))
+	}
+
+	if v, ok := d.GetOk("root_directory"); ok {
+		input.RootDirectory = expandEfsAccessPointRootDirectory(v.([]interface{}))
+	}
+
+	log.Printf("[DEBUG] Creating EFS Access Point: %#v", input)
+
+	ap, err := conn.CreateAccessPoint(&input)
+	if err != nil {
+		return err
+	}
+
+	d.SetId(*ap.AccessPointId)
+	log.Printf("[INFO] EFS access point ID: %s", d.Id())
+
+	stateConf := &resource.StateChangeConf{
+		Pending: []string{efs.LifeCycleStateCreating},
+		Target:  []string{efs.LifeCycleStateAvailable},
+		Refresh: func() (interface{}, string, error) {
+			resp, err := conn.DescribeAccessPoints(&efs.DescribeAccessPointsInput{
+				AccessPointId: aws.String(d.Id()),
+			})
+			if err != nil {
+				return nil, "error", err
+			}
+
+			if hasEmptyAccessPoints(resp) {
+				return nil, "error", fmt.Errorf("EFS access point %q could not be found.", d.Id())
+			}
+
+			mt := resp.AccessPoints[0]
+
+			log.Printf("[DEBUG] Current status of %q: %q", *mt.AccessPointId, *mt.LifeCycleState)
+			return mt, *mt.LifeCycleState, nil
```

Comment:
To prevent potential panics, we should opt to use the AWS Go SDK conversion functions here:

```suggestion
			log.Printf("[DEBUG] Current status of %q: %q", aws.StringValue(mt.AccessPointId), aws.StringValue(mt.LifeCycleState))
			return mt, aws.StringValue(mt.LifeCycleState), nil
```